File: Dataset/Leetcode/valid/78/623.cpp
 class Solution {
public:
    vector<vector<int>> res;
    void tra(vector<int>& nums,int n,int i,vector<int> &tmp){
        if(i==n){
            res.push_back(tmp);
            return;
        }
        tmp.push_back(nums[i]);
        tra(nums,n,i+1,tmp);
        tmp.pop_back();
        tra(nums,n,i+1,tmp);
    }
    vector<vector<int>> XXX(vector<int>& nums) {
        res.clear();
        if(nums.size()==0) return res;
        vector<int> tmp;
        tra(nums,nums.size(),0,tmp); 
        return res;
    }
};

File: Dataset/Leetcode/train/62/623.java
class Solution {
    public int XXX(int m, int n) {
        int[] dp = new int[n + 1];
        dp[1] = 1;
        for (int i = 1; i <= m; i++) {
            for (int j = 1; j <= n; j++) {
                dp[j] = dp[j - 1] + dp[j];
            }
        }
        return dp[n];
    }
}

